packages/core/CalendarManager.ts

```diff
@@ -217,7 +217,8 @@ export const getBusyCalendarTimes = async (
 
 export const createEvent = async (
   credential: CredentialWithAppName,
-  calEvent: CalendarEvent
+  calEvent: CalendarEvent,
+  externalId?: string
```

Comment:
I had a hard time figuring out how to pass externalId here, so I opted for this approach as it worn't interfere with createEvent calls that don't send this new parameter.

Reply:
Basically here we have 3 types of data eventCalendar, resultEvent and credential and I find that if I added the anchor value on resultEvent it wont have that many side effects in the entire process so it can be passed properly without trouble.